app/helpers/course_summaries_helper.rb

```diff
@@ -0,0 +1,70 @@
+module CourseSummariesHelper
+
+	###
+	# Get JSON data for the table in the form of
+	# { id: student_id, user_name: student.user_name,
+	#   first_name: student.first_name, last_name: student.last_name
+	#   marks: [ // an array of marks for each assignment, null if no grade ] }
+	###
+	def get_table_json_data()
+		all_students = Student.where(type: 'Student');
+		all_assignments = Assignment.all();
+
+		studentList = all_students.map do |student|
+			{
+				:id => student.id,
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.<br>Tab detected.